### ISSUE TITLE
How does one actually do batch processing?

### ISSUE THREAD

**Issue by michaelgroeger** [jacobgil/pytorch-grad-cam#264]
Hello @jacobgil, 

Thank you very much for this nice library. I find it very helpful so far. 

However, I can't figure out how to use batched inputs. To make a very simple PoC I played around with the semantic segmentation tutorial you provided, since this is my use case. 

I simply doubled the input to a batch of two images to see if I get the same activations for both images given the batch. Unfortunately, `grayscale_cam` has the same shape as the input batch, that is `(2, 375, 500)`, but only the first element of the batch contains activations while the second one is all zeros. 

I did also try to run `cam.batch_size = 2` but it didn't work. 

Maybe I also understood the documentation wrong. In the Readme it is written that "High performance: full support for batches of images in all methods.", but in cam.py line 113 it says "# AblationCAM and ScoreCAM have batched implementations.". 

I would immensely appreciate any help. 

P.S.: I also attached the code I played around with 

`
import warnings
warnings.filterwarnings('ignore')
warnings.simplefilter('ignore')
from torchvision.models.segmentation import deeplabv3_resnet50
import torch
import torch.functional as F
import numpy as np
import requests
import torchvision
from PIL import Image
from pytorch_grad_cam.utils.image import show_cam_on_image, preprocess_image

image_url = "https://farm1.staticflickr.com/6/9606553_ccc7518589_z.jpg"
image = np.array(Image.open(requests.get(image_url, stream=True).raw))
rgb_img = np.float32(image) / 255
input_tensor = preprocess_image(rgb_img,
                                mean=[0.485, 0.456, 0.406],
                                std=[0.229, 0.224, 0.225])
# Taken from the torchvision tutorial
# https://pytorch.org/vision/stable/auto_examples/plot_visualization_utils.html
model = deeplabv3_resnet50(pretrained=True, progress=False)
model = model.eval()

if torch.cuda.is_available():
    model = model.cuda()
    input_tensor = input_tensor.cuda()

output = model(input_tensor)
print(type(output), output.keys())

class SegmentationModelOutputWrapper(torch.nn.Module):
    def __init__(self, model): 
        super(SegmentationModelOutputWrapper, self).__init__()
        self.model = model
        
    def forward(self, x):
        return self.model(x)["out"]
    
model = SegmentationModelOutputWrapper(model)
output = model(input_tensor)

normalized_masks = torch.nn.functional.softmax(output, dim=1).cpu()
sem_classes = [
    '__background__', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus',
    'car', 'cat', 'chair', 'cow', 'diningtable', 'dog', 'horse', 'motorbike',
    'person', 'pottedplant', 'sheep', 'sofa', 'train', 'tvmonitor'
]
sem_class_to_idx = {cls: idx for (idx, cls) in enumerate(sem_classes)}

car_category = sem_class_to_idx["car"]
car_mask = normalized_masks[0, :, :, :].argmax(axis=0).detach().cpu().numpy()
car_mask_uint8 = 255 * np.uint8(car_mask == car_category)
car_mask_float = np.float32(car_mask == car_category)

both_images = np.hstack((image, np.repeat(car_mask_uint8[:, :, None], 3, axis=-1)))
Image.fromarray(both_images)

base = torch.zeros([2, 3, 375, 500])
base[0] = input_tensor
base[1] = input_tensor

from pytorch_grad_cam import GradCAM, ScoreCAM, AblationCAM

class SemanticSegmentationTarget:
    def __init__(self, category, mask):
        self.category = category
        self.mask = torch.from_numpy(mask)
        if torch.cuda.is_available():
            self.mask = self.mask.cuda()
        
    def __call__(self, model_output):
        return (model_output[self.category, :, : ] * self.mask).sum()

    
target_layers = [model.model.backbone.layer4]
targets = [SemanticSegmentationTarget(car_category, car_mask_float)]
with GradCAM(model=model,
             target_layers=target_layers,
             use_cuda=torch.cuda.is_available()) as cam:
    cam.batch_size = 2
    grayscale_cam = cam(input_tensor=base,
                        targets=targets)
    print(grayscale_cam.shape)
    print(np.unique(grayscale_cam[0]))
    print(np.unique(grayscale_cam[1]))
    grayscale_cam = grayscale_cam[1,:]
    cam_image = show_cam_on_image(rgb_img, grayscale_cam, use_rgb=True)

Image.fromarray(cam_image)
`

**Comment by jacobgil** [comment]
Hi,
The targets need to be specified for every member in the batch.
So 
targets = [SemanticSegmentationTarget(car_category, car_mask_float)]
Needs to be
targets = [SemanticSegmentationTarget(car_category, car_mask_float), SemanticSegmentationTarget(car_category, car_mask_float)]

Can you please try this and tell if it worked?
